### ISSUE TITLE
upgrading PL via pip uninstalls pytorch-1.8

### ISSUE THREAD

**Issue by stas00** [PyTorchLightning/pytorch-lightning#4596]
### To Reproduce

```
$ pip install pytorch-lightning -U
[...]
Installing collected packages: torch, pytorch-lightning
  Attempting uninstall: torch
    Found existing installation: torch 1.8.0.dev20201106+cu110
    Uninstalling torch-1.8.0.dev20201106+cu110:
      Successfully uninstalled torch-1.8.0.dev20201106+cu110
```

### Expected behavior

It shouldn't uninstall pytorch-1.8.

Nvidia RTX-30* don't work with pytorch < 1.8.

### Environment

```
* CUDA:
        - GPU:
                - GeForce RTX 3090
                - GeForce GTX 1070 Ti
        - available:         True
        - version:           10.2
* Packages:
        - numpy:             1.19.4
        - pyTorch_debug:     True
        - pyTorch_version:   1.7.0
        - pytorch-lightning: 1.0.5
        - tqdm:              4.50.2
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.8.5
        - version:           #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020
```

**Comment by stas00** [comment]
This is weird, the report says I still had pytorch-1.7 after removing pytorch-1.8, how would 2 of them co-exist in the same conda setup. But still PL works just fine with pt-nightly, so it probably shouldn't force <1.8, right?

**Comment by Borda** [comment]
@stas00 I think it is a problem with your setting `-U` which upgraded the package along with all its dependencies
Talking about `<1.8` it is there as we do not have any testing for nightly 1.8 yet, but most likely it shall be fine and there is no special need for such limitation, @edenlightning?

**Comment by edenlightning** [comment]
not that I am aware.

Is there any bug on lightning then?

**Comment by stas00** [comment]
> @stas00 I think it is a problem with your setting -U which upgraded the package along with all its dependencies

That's what I'm saying - is there a reason to force `torch<1.8`? Then pip won't remove it. Is my thinking correct?

**Comment by stas00** [comment]
Found another problem with this:

We recently added a PL version check to handle breaking changes in PL, 
https://github.com/huggingface/transformers/blob/master/examples/lightning_base.py#L36
and it doesn't look it's what we want, since I get:
```
$ python -c 'import pkg_resources; pkg_resources.require("pytorch_lightning>=1.0.4")'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/stas/anaconda3/envs/main-38/lib/python3.8/site-packages/pkg_resources/__init__.py", line 884, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/stas/anaconda3/envs/main-38/lib/python3.8/site-packages/pkg_resources/__init__.py", line 775, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (torch 1.8.0.dev20201106+cu110 (/mnt/nvme1/anaconda3/envs/main-38/lib/python3.8/site-packages), Requirement.parse('torch<1.8,>=1.3'), {'pytorch-lightning'})
```
So it fails despite having 1.0.5 installed.

Which way would you recommend we check for the PL version - w/o the above issue? I see there is `pytorch_lightning.__version__` - i.e. is this public API we can rely on? Thanks.

**Comment by s-rog** [comment]
When I'm using non release versions of pytorch I install lightning and other torch depending libraries with `--no-deps` and haven't had any issues, though I guess it would be problematic if your env is very barebones...

**Comment by stas00** [comment]
given that `conda` has been choking on itself as of recent with just a handful of packages and their dozens of dependencies, I end up re-installing the env from scratch very often :( So tracking down all the dependencies would be a job on its own.

So the bare-bones setup to run `transformers` dev I have to do:
```
cd /hf/transformers-master
pip install -e .[dev]
pip install -r examples/requirements.txt
```
which pulls in PL and its deps.

And since there is no proper support for RTX-30 series yet, I have to use nightlies for tf/pt:
```
pip install tf-nightly-gpu tensorflow-datasets -U
pip install --pre torch torchvision -f https://download.pytorch.org/whl/nightly/cu110/torch_nightly.html -U
```
that's all I have in my env (and a few jupyter packages) - which still is a lot of deps.

**Comment by Borda** [comment]
@stas00 FYI, we have fixed the version freeze on last 1.0.6 for PyPI, and for Conda, it shall be fine for all versions...

**Comment by stas00** [comment]
That's wonderful, thank you, @Borda!
